### PR TITLE
Migrate to CircleCI v2

### PR DESCRIPTION
**Pull request by rocioar**
Migrating to CircleCI version 2.

JIRA Ticket: OC-4053

Testing instructions:

Check that tests are passing.

Note to reviewer:

- Had to upgrade to python version 2.7.14, since 2.7.10 is not in the list of docker images: https://github.com/CircleCI-Public/circleci-dockerfiles/tree/master/python/images.
- Installing firefox, since it doesn't come with the environment anymore.
- Added cache for python requirements

Reviewers:

[Sven]

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -1,27 +1,50 @@
...
+version: 2
+jobs:
+  build:
+    docker:
+      - image: circleci/python:2.7.14
+    parallelism: 4
+    steps:
+      - checkout
+      - restore_cache:
+          key: deps1-{{ .Branch }}-{{ checksum "requirements.txt" }}-{{ checksum "test_requirements.txt" }}
```
Comment (smarnach) [inline circle.yml]: We are also installing several pacakges directly from this file (`circle.yml`): pip, wheel, setuptools, xblock-sdk, the requirements of xblock-sdk.  So we should include the checksum of `circle.yml` in the cache key as well.

File: circle.yml [hunk 2]
```diff
@@ -1,27 +1,50 @@
...
+            - "venv"
+      - run:
+          name: Install firefox 38.0.5
+          command: |
+            FIREFOX_URL="https://sourceforge.net/projects/ubuntuzilla/files/mozilla/apt/pool/main/f/firefox-mozilla-build/firefox-mozilla-build_38.0.5-0ubuntu1_amd64.deb/download" \
+            && curl --silent --show-error --location --fail --retry 3 --output /tmp/firefox.deb $FIREFOX_URL \
+            && sudo dpkg -i /tmp/firefox.deb || sudo apt-get -f install  \
+            && sudo apt-get install -y libgtk3.0-cil-dev libasound2 libasound2 libdbus-glib-1-2 libdbus-1-3 libgtk2.0-0 \
+            && rm -rf /tmp/firefox.deb \
+            && firefox --version
```
Comment (smarnach) [inline circle.yml]: This looks rather awkward.  Isn't there an Ubuntu PPA we can install the right version of Firefox from?  It's fine to leave it as it is, since you figured it out already. :)

File: circle.yml [hunk 3]
```diff
@@ -1,27 +1,50 @@
...
+      - run:
+          name: Run tests
+          environment:
+            DISPLAY: :99
+          command: |
+            sudo Xvfb :99 -screen 0 1280x1024x24 & exec "$@"
+            . venv/bin/activate
+            if [ $CIRCLE_NODE_INDEX = '0' ]; then pep8 problem_builder --max-line-length=120; fi
```
Comment (smarnach) [inline circle.yml]: For this to make sense, we need to configure parallelism as well – I think it's currently only using a single container.

File: circle.yml [hunk 4]
```diff
@@ -1,27 +1,50 @@
...
+            && sudo dpkg -i /tmp/firefox.deb || sudo apt-get -f install  \
+            && sudo apt-get install -y libgtk3.0-cil-dev libasound2 libasound2 libdbus-glib-1-2 libdbus-1-3 libgtk2.0-0 \
+            && rm -rf /tmp/firefox.deb \
+            && firefox --version
+      - run:
+          name: Run tests
+          environment:
+            DISPLAY: :99
+          command: |
+            sudo Xvfb :99 -screen 0 1280x1024x24 & exec "$@"
```
Comment (smarnach) [inline circle.yml]: Why is this required now, while it wasn't required in the old version?  (Just trying to understand what's going on.)

File: circle.yml (hunk 1 above)
Comment (rocioar) [reply circle.yml]: good point 👍  will fix.

File: circle.yml (hunk 4 above)
Comment (rocioar) [reply circle.yml]: here is the explanation: https://circleci.com/docs/1.0/build-image-precise/#browsers

File: circle.yml (hunk 2 above)
Comment (rocioar) [reply circle.yml]: I couldn't find a ppa with this version of firefox (38), for other versions the test fail.. or some incompatibility with Selenium is raised.

**Comment by rocioar** [comment]
@smarnach I've addressed your comments, and the PR is ready for review again. Thanks as usual for the in-depth review.

File: circle.yml (hunk 4 above)
Comment (smarnach) [reply circle.yml]: Thanks for looking this up.  Seems like the new version of CircleCI requires us to be a lot more explicit about several things.

I still don't fully understand what's happening in this line.  Why is Xvfb run as root? I always run it as a normal user.  It's also unclear what `exec "$@"` does here – why doesn't this result in an infinite recursion?

I usually simply wrap the command that needs access to the fake X server in `xvfb-run`, e.g.

    xvfb-run --server-args="-screen 0 1280x1024x24" python run_tests.py ${TESTFILES}

`xvfb-run` automatically sets the `DISPLAY` variable.  Maybe you can try this option as well?  There isn't really anything wrong with what you have, but I prefer code that I can understand. :)

File: circle.yml (hunk 2 above)
Comment (smarnach) [reply circle.yml]: OK, fair enough.  Let's just stick with what you have for now.

File: circle.yml (hunk 3 above)
Comment (smarnach) [reply circle.yml]: Sorry, I missed that.  All good then.
